### PR TITLE
Added generic MAAT API service class

### PR DESCRIPTION
**Pull request by tanovellino**
Added a first markup of a service class to manage Matt API calls

## What

[Link to story](https://dsdmoj.atlassian.net/browse/LASB-XXX)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- [x] Tests should be passing: `./gradlew test`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Comment by tanovellino** [comment]
This first pull request is mainly jus to have it on cloud for sharing and discussion

File: dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/mattapi/MattApiService.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+import reactor.core.publisher.Mono;
+
+import java.util.Map;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class MattApiService {
```
Comment (atharmajeed) [inline dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/mattapi/MattApiService.java:18]: Its spells as a MAAT

Comment (tanovellino) [reply dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/mattapi/MattApiService.java:17]: I noticed. Thanks. It is changed now

**Review by atharmajeed — APPROVED** [review]
Looks good and happy to approve. There are a few minor codes smells that might need to look up again. You can check the Sonar

**Pull request merged**